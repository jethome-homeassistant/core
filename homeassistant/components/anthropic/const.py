"""Constants for the Anthropic integration."""

import logging

DOMAIN = "anthropic"
LOGGER = logging.getLogger(__package__)

DEFAULT_CONVERSATION_NAME = "Claude conversation"

CONF_RECOMMENDED = "recommended"
CONF_PROMPT = "prompt"
CONF_CHAT_MODEL = "chat_model"
RECOMMENDED_CHAT_MODEL = "claude-3-5-haiku-latest"
CONF_MAX_TOKENS = "max_tokens"
RECOMMENDED_MAX_TOKENS = 3000
CONF_TEMPERATURE = "temperature"
RECOMMENDED_TEMPERATURE = 1.0
CONF_THINKING_BUDGET = "thinking_budget"
RECOMMENDED_THINKING_BUDGET = 0
MIN_THINKING_BUDGET = 1024
CONF_WEB_SEARCH = "web_search"
RECOMMENDED_WEB_SEARCH = False
CONF_WEB_SEARCH_USER_LOCATION = "user_location"
RECOMMENDED_WEB_SEARCH_USER_LOCATION = False
CONF_WEB_SEARCH_MAX_USES = "web_search_max_uses"
RECOMMENDED_WEB_SEARCH_MAX_USES = 5
CONF_WEB_SEARCH_CITY = "city"
CONF_WEB_SEARCH_REGION = "region"
CONF_WEB_SEARCH_COUNTRY = "country"
CONF_WEB_SEARCH_TIMEZONE = "timezone"

NON_THINKING_MODELS = [
    "claude-3-5",  # Both sonnet and haiku
    "claude-3-opus",
    "claude-3-haiku",
]

WEB_SEARCH_UNSUPPORTED_MODELS = [
    "claude-3-haiku",
    "claude-3-opus",
    "claude-3-5-sonnet-20240620",
    "claude-3-5-sonnet-20241022",
]
