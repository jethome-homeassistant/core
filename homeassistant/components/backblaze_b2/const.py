"""Constants for the Backblaze B2 integration."""

from collections.abc import Callable
from typing import Final

from homeassistant.util.hass_dict import HassKey

DOMAIN: Final = "backblaze_b2"

CONF_KEY_ID = "key_id"
CONF_APPLICATION_KEY = "application_key"
CONF_BUCKET = "bucket"
CONF_PREFIX = "prefix"

DATA_BACKUP_AGENT_LISTENERS: HassKey[list[Callable[[], None]]] = HassKey(
    f"{DOMAIN}.backup_agent_listeners"
)

METADATA_FILE_SUFFIX = ".metadata.json"
METADATA_VERSION = "1"

BACKBLAZE_REALM = "production"
