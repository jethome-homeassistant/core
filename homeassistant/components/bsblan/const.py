"""Constants for the BSB-Lan integration."""

from __future__ import annotations

from datetime import timedelta
import logging
from typing import Final

# Integration domain
DOMAIN: Final = "bsblan"

LOGGER = logging.getLogger(__package__)
SCAN_INTERVAL = timedelta(seconds=12)  # Legacy interval, kept for compatibility
SCAN_INTERVAL_FAST = timedelta(seconds=12)  # For state/sensor data
SCAN_INTERVAL_SLOW = timedelta(minutes=5)  # For config data

# Services
DATA_BSBLAN_CLIENT: Final = "bsblan_client"

ATTR_TARGET_TEMPERATURE: Final = "target_temperature"
ATTR_INSIDE_TEMPERATURE: Final = "inside_temperature"
ATTR_OUTSIDE_TEMPERATURE: Final = "outside_temperature"

CONF_PASSKEY: Final = "passkey"

DEFAULT_PORT: Final = 80
