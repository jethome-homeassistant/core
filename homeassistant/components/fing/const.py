"""Const for the Fing integration."""

DOMAIN = "fing"
UPNP_AVAILABLE = "upnp_available"
