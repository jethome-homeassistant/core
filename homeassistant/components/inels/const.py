"""Constants for the iNELS integration."""

import logging

from homeassistant.const import Platform

DOMAIN = "inels"
TITLE = "iNELS"

PLATFORMS: list[Platform] = [
    Platform.SWITCH,
]

LOGGER = logging.getLogger(__package__)
