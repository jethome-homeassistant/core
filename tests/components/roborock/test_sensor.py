"""Test Roborock Sensors."""

from unittest.mock import patch

import pytest
from roborock import DeviceData, HomeDataDevice
from roborock.roborock_message import RoborockMessage, RoborockMessageProtocol
from roborock.version_1_apis import RoborockMqttClientV1
from syrupy.assertion import SnapshotAssertion

from homeassistant.const import Platform
from homeassistant.core import HomeAssistant

from .mock_data import CONSUMABLE, STATUS, USER_DATA

from tests.common import MockConfigEntry


@pytest.fixture
def platforms() -> list[Platform]:
    """Fixture to set platforms used in the test."""
    return [Platform.SENSOR]


async def test_sensors(
    hass: HomeAssistant,
    setup_entry: MockConfigEntry,
    snapshot: SnapshotAssertion,
) -> None:
    """Test sensors and check test values are correctly set."""
    assert snapshot == hass.states.async_all("sensor")


async def test_listener_update(
    hass: HomeAssistant, setup_entry: MockConfigEntry
) -> None:
    """Test that when we receive a mqtt topic, we successfully update the entity."""
    assert hass.states.get("sensor.roborock_s7_maxv_status").state == "charging"
    # Listeners are global based on uuid - so this is okay
    client = RoborockMqttClientV1(
        USER_DATA, DeviceData(device=HomeDataDevice("abc123", "", "", "", ""), model="")
    )
    # Test Status
    with patch("roborock.version_1_apis.AttributeCache.value", STATUS.as_dict()):
        # Symbolizes a mqtt message coming in
        client.on_message_received(
            [
                RoborockMessage(
                    protocol=RoborockMessageProtocol.GENERAL_REQUEST,
                    payload=b'{"t": 1699464794, "dps": {"121": 5}}',
                )
            ]
        )
    # Test consumable
    assert (
        hass.states.get("sensor.roborock_s7_maxv_filter_time_left").state
        == "129.338333333333"
    )
    with patch("roborock.version_1_apis.AttributeCache.value", CONSUMABLE.as_dict()):
        client.on_message_received(
            [
                RoborockMessage(
                    protocol=RoborockMessageProtocol.GENERAL_REQUEST,
                    payload=b'{"t": 1699464794, "dps": {"127": 743}}',
                )
            ]
        )
    await hass.async_block_till_done()
    assert (
        hass.states.get("sensor.roborock_s7_maxv_filter_time_left").state
        == "149.793611111111"
    )
