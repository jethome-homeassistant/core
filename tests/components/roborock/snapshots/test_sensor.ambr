# serializer version: 1
# name: test_sensors
  list([
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Main brush time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_main_brush_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '279.338333333333',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Side brush time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_side_brush_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '179.338333333333',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Filter time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_filter_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '129.338333333333',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 MaxV Dock Maintenance brush time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_dock_maintenance_brush_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '235',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 MaxV Dock Strainer time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_dock_strainer_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '85',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Sensor time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_sensor_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '9.33833333333333',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Cleaning time',
        'unit_of_measurement': <UnitOfTime.MINUTES: 'min'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_cleaning_time',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '19.6',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Total cleaning time',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_total_cleaning_time',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '20.6616666666667',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Roborock S7 2 Total cleaning count',
        'state_class': <SensorStateClass.TOTAL_INCREASING: 'total_increasing'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_total_cleaning_count',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '31',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Roborock S7 2 Status',
        'options': list([
          'unknown',
          'starting',
          'charger_disconnected',
          'idle',
          'remote_control_active',
          'cleaning',
          'returning_home',
          'manual_mode',
          'charging',
          'charging_problem',
          'paused',
          'spot_cleaning',
          'error',
          'shutting_down',
          'updating',
          'docking',
          'going_to_target',
          'zoned_cleaning',
          'segment_cleaning',
          'emptying_the_bin',
          'washing_the_mop',
          'washing_the_mop_2',
          'going_to_wash_the_mop',
          'in_call',
          'mapping',
          'egg_attack',
          'patrol',
          'attaching_the_mop',
          'detaching_the_mop',
          'charging_complete',
          'device_offline',
          'locked',
          'air_drying_stopping',
          'robot_status_mopping',
          'clean_mop_cleaning',
          'clean_mop_mopping',
          'segment_mopping',
          'segment_clean_mop_cleaning',
          'segment_clean_mop_mopping',
          'zoned_mopping',
          'zoned_clean_mop_cleaning',
          'zoned_clean_mop_mopping',
          'back_to_dock_washing_duster',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_status',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'charging',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Roborock S7 2 Cleaning area',
        'unit_of_measurement': <UnitOfArea.SQUARE_METERS: 'm²'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_cleaning_area',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '21.0',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Roborock S7 2 Total cleaning area',
        'unit_of_measurement': <UnitOfArea.SQUARE_METERS: 'm²'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_total_cleaning_area',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '1159.2',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Roborock S7 2 Vacuum error',
        'options': list([
          'none',
          'lidar_blocked',
          'bumper_stuck',
          'wheels_suspended',
          'cliff_sensor_error',
          'main_brush_jammed',
          'side_brush_jammed',
          'wheels_jammed',
          'robot_trapped',
          'no_dustbin',
          'strainer_error',
          'compass_error',
          'low_battery',
          'charging_error',
          'battery_error',
          'wall_sensor_dirty',
          'robot_tilted',
          'side_brush_error',
          'fan_error',
          'dock',
          'optical_flow_sensor_dirt',
          'vertical_bumper_pressed',
          'dock_locator_error',
          'return_to_dock_fail',
          'nogo_zone_detected',
          'visual_sensor',
          'light_touch',
          'vibrarise_jammed',
          'robot_on_carpet',
          'filter_blocked',
          'invisible_wall_detected',
          'cannot_cross_carpet',
          'internal_error',
          'collect_dust_error_3',
          'collect_dust_error_4',
          'mopping_roller_1',
          'mopping_roller_error_2',
          'clear_water_box_hoare',
          'dirty_water_box_hoare',
          'sink_strainer_hoare',
          'clear_water_box_exception',
          'clear_brush_exception',
          'clear_brush_exception_2',
          'filter_screen_exception',
          'mopping_roller_2',
          'up_water_exception',
          'drain_water_exception',
          'temperature_protection',
          'clean_carousel_exception',
          'clean_carousel_water_full',
          'water_carriage_drop',
          'check_clean_carouse',
          'audio_error',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_vacuum_error',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'none',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'battery',
        'friendly_name': 'Roborock S7 2 Battery',
        'unit_of_measurement': '%',
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_battery',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '100',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'timestamp',
        'friendly_name': 'Roborock S7 2 Last clean begin',
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_last_clean_begin',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '2023-01-01T03:22:10+00:00',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'timestamp',
        'friendly_name': 'Roborock S7 2 Last clean end',
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_last_clean_end',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '2023-01-01T03:43:58+00:00',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Roborock S7 MaxV Dock Dock error',
        'options': list([
          'ok',
          'duct_blockage',
          'water_empty',
          'waste_water_tank_full',
          'maintenance_brush_jammed',
          'dirty_tank_latch_open',
          'no_dustbin',
          'cleaning_tank_full_or_blocked',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_dock_dock_error',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'ok',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Main brush time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_main_brush_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '279.338333333333',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Side brush time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_side_brush_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '179.338333333333',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Filter time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_filter_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '129.338333333333',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Dock Maintenance brush time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_dock_maintenance_brush_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '235',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Dock Strainer time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_dock_strainer_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '85',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Sensor time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_sensor_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '9.33833333333333',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Cleaning time',
        'unit_of_measurement': <UnitOfTime.MINUTES: 'min'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_cleaning_time',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '19.6',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Roborock S7 2 Total cleaning time',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_total_cleaning_time',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '20.6616666666667',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Roborock S7 2 Total cleaning count',
        'state_class': <SensorStateClass.TOTAL_INCREASING: 'total_increasing'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_total_cleaning_count',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '31',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Roborock S7 2 Status',
        'options': list([
          'unknown',
          'starting',
          'charger_disconnected',
          'idle',
          'remote_control_active',
          'cleaning',
          'returning_home',
          'manual_mode',
          'charging',
          'charging_problem',
          'paused',
          'spot_cleaning',
          'error',
          'shutting_down',
          'updating',
          'docking',
          'going_to_target',
          'zoned_cleaning',
          'segment_cleaning',
          'emptying_the_bin',
          'washing_the_mop',
          'washing_the_mop_2',
          'going_to_wash_the_mop',
          'in_call',
          'mapping',
          'egg_attack',
          'patrol',
          'attaching_the_mop',
          'detaching_the_mop',
          'charging_complete',
          'device_offline',
          'locked',
          'air_drying_stopping',
          'robot_status_mopping',
          'clean_mop_cleaning',
          'clean_mop_mopping',
          'segment_mopping',
          'segment_clean_mop_cleaning',
          'segment_clean_mop_mopping',
          'zoned_mopping',
          'zoned_clean_mop_cleaning',
          'zoned_clean_mop_mopping',
          'back_to_dock_washing_duster',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_status',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'charging',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Roborock S7 2 Cleaning area',
        'unit_of_measurement': <UnitOfArea.SQUARE_METERS: 'm²'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_cleaning_area',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '21.0',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'friendly_name': 'Roborock S7 2 Total cleaning area',
        'unit_of_measurement': <UnitOfArea.SQUARE_METERS: 'm²'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_total_cleaning_area',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '1159.2',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Roborock S7 2 Vacuum error',
        'options': list([
          'none',
          'lidar_blocked',
          'bumper_stuck',
          'wheels_suspended',
          'cliff_sensor_error',
          'main_brush_jammed',
          'side_brush_jammed',
          'wheels_jammed',
          'robot_trapped',
          'no_dustbin',
          'strainer_error',
          'compass_error',
          'low_battery',
          'charging_error',
          'battery_error',
          'wall_sensor_dirty',
          'robot_tilted',
          'side_brush_error',
          'fan_error',
          'dock',
          'optical_flow_sensor_dirt',
          'vertical_bumper_pressed',
          'dock_locator_error',
          'return_to_dock_fail',
          'nogo_zone_detected',
          'visual_sensor',
          'light_touch',
          'vibrarise_jammed',
          'robot_on_carpet',
          'filter_blocked',
          'invisible_wall_detected',
          'cannot_cross_carpet',
          'internal_error',
          'collect_dust_error_3',
          'collect_dust_error_4',
          'mopping_roller_1',
          'mopping_roller_error_2',
          'clear_water_box_hoare',
          'dirty_water_box_hoare',
          'sink_strainer_hoare',
          'clear_water_box_exception',
          'clear_brush_exception',
          'clear_brush_exception_2',
          'filter_screen_exception',
          'mopping_roller_2',
          'up_water_exception',
          'drain_water_exception',
          'temperature_protection',
          'clean_carousel_exception',
          'clean_carousel_water_full',
          'water_carriage_drop',
          'check_clean_carouse',
          'audio_error',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_vacuum_error',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'none',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'battery',
        'friendly_name': 'Roborock S7 2 Battery',
        'unit_of_measurement': '%',
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_battery',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '100',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'timestamp',
        'friendly_name': 'Roborock S7 2 Last clean begin',
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_last_clean_begin',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '2023-01-01T03:22:10+00:00',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'timestamp',
        'friendly_name': 'Roborock S7 2 Last clean end',
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_last_clean_end',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '2023-01-01T03:43:58+00:00',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Roborock S7 2 Dock Dock error',
        'options': list([
          'ok',
          'duct_blockage',
          'water_empty',
          'waste_water_tank_full',
          'maintenance_brush_jammed',
          'dirty_tank_latch_open',
          'no_dustbin',
          'cleaning_tank_full_or_blocked',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_dock_dock_error',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'ok',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Roborock S7 2 Current room',
        'options': list([
          'Example room 1',
          'Example room 2',
          'Example room 3',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_maxv_current_room',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'Example room 2',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Roborock S7 2 Current room',
        'options': list([
          'Example room 1',
          'Example room 2',
          'Example room 3',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.roborock_s7_2_current_room',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'Example room 2',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Dyad Pro Status',
        'options': list([
          'unknown',
          'fetching',
          'fetch_failed',
          'updating',
          'washing',
          'ready',
          'charging',
          'mop_washing',
          'self_clean_cleaning',
          'self_clean_deep_cleaning',
          'self_clean_rinsing',
          'self_clean_dehydrating',
          'drying',
          'ventilating',
          'reserving',
          'mop_washing_paused',
          'dusting_mode',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.dyad_pro_status',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'drying',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'battery',
        'friendly_name': 'Dyad Pro Battery',
        'unit_of_measurement': '%',
      }),
      'context': <ANY>,
      'entity_id': 'sensor.dyad_pro_battery',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '100',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Dyad Pro Filter time left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.dyad_pro_filter_time_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '0.0308333333333333',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Dyad Pro Roller left',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.dyad_pro_roller_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '0.0616666666666667',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Dyad Pro Error',
        'options': list([
          'none',
          'dirty_tank_full',
          'water_level_sensor_stuck',
          'clean_tank_empty',
          'clean_head_entangled',
          'clean_head_too_hot',
          'fan_protection_e5',
          'cleaning_head_blocked',
          'temperature_protection',
          'fan_protection_e4',
          'fan_protection_e9',
          'battery_temperature_protection_e0',
          'battery_temperature_protection',
          'battery_temperature_protection_2',
          'power_adapter_error',
          'dirty_charging_contacts',
          'low_battery',
          'battery_under_10',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.dyad_pro_error',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'none',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Dyad Pro Total cleaning time',
        'unit_of_measurement': <UnitOfTime.HOURS: 'h'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.dyad_pro_total_cleaning_time',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '3.55',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Zeo One State',
        'options': list([
          'standby',
          'weighing',
          'soaking',
          'washing',
          'rinsing',
          'spinning',
          'drying',
          'cooling',
          'under_delay_start',
          'done',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.zeo_one_state',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'drying',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Zeo One Countdown',
        'unit_of_measurement': <UnitOfTime.MINUTES: 'min'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.zeo_one_countdown',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '0',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'duration',
        'friendly_name': 'Zeo One Washing left',
        'unit_of_measurement': <UnitOfTime.MINUTES: 'min'>,
      }),
      'context': <ANY>,
      'entity_id': 'sensor.zeo_one_washing_left',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': '253',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'device_class': 'enum',
        'friendly_name': 'Zeo One Error',
        'options': list([
          'none',
          'refill_error',
          'drain_error',
          'door_lock_error',
          'water_level_error',
          'inverter_error',
          'heating_error',
          'temperature_error',
          'communication_error',
          'drying_error',
          'drying_error_e_12',
          'drying_error_e_13',
          'drying_error_e_14',
          'drying_error_e_15',
          'drying_error_e_16',
          'drying_error_water_flow',
          'drying_error_restart',
          'spin_error',
        ]),
      }),
      'context': <ANY>,
      'entity_id': 'sensor.zeo_one_error',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'none',
    }),
  ])
# ---
